src/Tasks/Microsoft.NET.Build.Tasks/GetEmbeddedApphostPaths.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public sealed class GetEmbeddedApphostPaths : TaskBase
```

Comment:
These code is copied out of GenerateShim, However, there is only around 10 lines are duplicated. And there is no critical code that will need to keep in sync. If we want to share these code, the overhead will be more than 10 lines.